components/camel-solr/pom.xml

```diff
@@ -0,0 +1,180 @@
+            <artifactId>commons-codec</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.httpcomponents</groupId>
+            <artifactId>httpclient</artifactId>
+            <version>${httpclient4-version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.httpcomponents</groupId>
+            <artifactId>httpmime</artifactId>
+            <version>${httpclient4-version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.zookeeper</groupId>
+            <artifactId>zookeeper</artifactId>
+            <version>${solr-zookeeper-version}</version>
+            <exclusions>
+                <exclusion>
+                    <groupId>org.slf4j</groupId>
+                    <artifactId>slf4j-log4j12</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>log4j</groupId>
+                    <artifactId>log4j</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+
+        <!-- test dependencies -->
+        <dependency>
+            <groupId>org.apache.camel</groupId>
+            <artifactId>camel-test-junit5</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.solr</groupId>
+            <artifactId>solr-test-framework</artifactId>
+            <version>${solr-version}</version>
+            <scope>test</scope>
+            <exclusions>
+                <exclusion>
+                    <groupId>com.google.protobuf</groupId>
+                    <artifactId>protobuf-java</artifactId>
+                </exclusion>
+                <!-- exclude rome 0.9 which is not a valid Maven artifact -->
+                <!-- rome is not a needed dependency -->
+                <exclusion>
+                    <groupId>rome</groupId>
+                    <artifactId>rome</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>jdk.tools</groupId>
+                    <artifactId>jdk.tools</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+        <dependency>
+            <groupId>com.google.protobuf</groupId>
+            <artifactId>protobuf-java</artifactId>
+            <scope>test</scope>
+        </dependency>
+
+		<dependency>
+    		<groupId>org.apache.solr</groupId>
+    		<artifactId>solr-extraction</artifactId>
+            <version>${solr-version}</version>
+            <scope>test</scope>    
+		</dependency>
+
+        <dependency>
+            <groupId>org.eclipse.jetty</groupId>
+            <artifactId>jetty-server</artifactId>
+            <version>10.0.20</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.eclipse.jetty</groupId>
+            <artifactId>jetty-servlet</artifactId>
+            <version>10.0.20</version>
```

Comment:
From parent pom

Reply:
Solr internally uses jetty-server 10.0.20 for testing purposes. 
solr/versions.lock
---
org.eclipse.jetty:jetty-server:10.0.22 (7 constraints: d86da720)
org.eclipse.jetty:jetty-servlet:10.0.22 (2 constraints: 3213f360)
org.eclipse.jetty:jetty-servlets:10.0.22 (1 constraints: 67059040)
---

If i upate to parent pom version than tests are failing with errors.

[INFO] Running org.apache.camel.component.solr.[1mInitSolrEndpointTest[m
[INFO] [1;32mTests run: [0;1;32m16[m, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.849 s -- in org.apache.camel.component.solr.[1mInitSolrEndpointTest[m
[INFO] Running org.apache.camel.component.solr.[1mSolrAddBeanTest[m
[ERROR] [1;31mTests [0;1mrun: [0;1m1[m, Failures: 0, [1;31mErrors: [0;1;31m1[m, Skipped: 0, Time elapsed: 0.116 s[1;31m <<< FAILURE![m -- in org.apache.camel.component.solr.[1mSolrAddBeanTest[m
[ERROR] org.apache.camel.component.solr.SolrAddBeanTest -- Time elapsed: 0.116 s <<< ERROR!
java.lang.NoClassDefFoundError: org/eclipse/jetty/server/SessionIdManager
	at org.apache.camel.component.solr.JettySolrFactory.createJetty(JettySolrFactory.java:112)
	at org.apache.camel.component.solr.JettySolrFactory.createJettyTestFixture(JettySolrFactory.java:131)
	at org.apache.camel.component.solr.SolrFixtures.createSolrFixtures(SolrFixtures.java:94)
	at org.apache.camel.component.solr.SolrComponentTestSupport.beforeClass(SolrComponentTestSupport.java:93)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
Caused by: java.lang.ClassNotFoundException: org.eclipse.jetty.server.SessionIdManager
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:525)
	... 6 more

These dependencies are used only for internal component testing. Mey be we can stay on these? And wait for solr to update or maybe in future replace with testcontainers? For the moment solr developers still use this org.apache.solr.embedded.JettySolrRunner.JettySolrRunner (10.0.20).

Reply:
Maybe add a property in the parent pom specific for this dependency

Reply:
moved version to camel/parent/pom property "jetty-for-solr-version "